pkg/reloader/reloader.go

```diff
@@ -692,16 +693,24 @@ func RuntimeInfoURLFromBase(u *url.URL) *url.URL {
 
 var envRe = regexp.MustCompile(`\$\(([a-zA-Z_0-9]+)\)`)
 
-func expandEnv(b []byte) (r []byte, err error) {
+func expandEnv(logger log.Logger, b []byte, suppressEnvironmentVariablesExpansionErrors bool, configApplyErrorsPtr *prometheus.Counter) (r []byte, err error) {
 	r = envRe.ReplaceAllFunc(b, func(n []byte) []byte {
 		if err != nil {
 			return nil
 		}
+		m := n
 		n = n[2 : len(n)-1]
 
 		v, ok := os.LookupEnv(string(n))
 		if !ok {
-			err = errors.Errorf("found reference to unset environment variable %q", n)
+			(*configApplyErrorsPtr).Inc()
```

Comment:
I'd rather see a new metric to track this: either an expansion errors counter or a gauge (the latter probably makes more sense as a counter needs constant increases to trigger an alert).